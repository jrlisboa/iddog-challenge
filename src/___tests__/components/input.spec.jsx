import React from 'react';
import { shallow } from 'enzyme';
import Input from '../../components/input';

import {
    InputStyle,
} from '../../components/input/index.style';

describe('Input component', () => {
    const event = {
        target: {
            name: 'myInput',
            value: 'foo',
        },
    };
    const onChangeMock = jest.fn((e) => {
        return e;
    });
    const component = shallow(
        <Input
            name="myInput"
            onChange={onChangeMock} />,
    );

    it('Should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('Should Input change value', () => {
        component.find(InputStyle).simulate('change', event);
        expect(onChangeMock).toHaveReturnedWith(event);
    });
});
