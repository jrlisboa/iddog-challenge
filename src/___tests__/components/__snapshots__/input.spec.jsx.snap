// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should match snapshot 1`] = `ShallowWrapper {}`;
